src/transformers/generation/utils.py

```diff
@@ -3365,34 +3354,80 @@ def _beam_search(
             `return_dict_in_generate=True` or a [`~generation.GenerateBeamEncoderDecoderOutput`] if
             `model.config.is_encoder_decoder=True`.
         """
-        # init values
+
+        def flatten_beam_dim(tensor):
+            """3D with beams in a separate dim -> 2D with batch*beam in batch dim"""
+            shape = list(tensor.shape)
+            return torch.reshape(tensor, [shape[0] * shape[1]] + shape[2:])
```

Comment:
not sure the comment is helpful, better to say that the input should be of dimension x, y, z and we make it x * y +z, and maybe whyu.